opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/MultiFieldQuery.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.opensearch.storage.script.filter.lucene.relevance;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.sql.expression.NamedArgumentExpression;
+
+/**
+ *  Base class to represent relevance queries that search multiple fields.
+ * @param <T> The builder class for the OpenSearch query.
+ */
+abstract class MultiFieldQuery<T extends QueryBuilder> extends RelevanceQuery<T> {
+
+  public MultiFieldQuery(String queryName, Map<String, QueryBuilderStep<T>> queryBuildActions) {
+    super(queryName, queryBuildActions);
```

Comment:
Instead of passing the queryName as a constructor parameter, you could create an abstract method and retrieve it here.  Each child would override the return value (returning the NAME).